### PR TITLE
Improvements job service

### PR DESCRIPTION
**Pull request by DanielElisenberg**
# IMPROVEMENTS JOB SERVICE

Improvements to datamodelling and responseformat on dates. Updates to docker-compose for mongodb instance.

### What's changed?
* Use standard .isoformat() to stringify ISOdates stored in db
* In addition to the upsert query, the jobs are now split into 2 collections
    * inprogress: asynchronous jobs. This collection has a unique index on datasetName to avoid importing a dataset twice
    * completed: Completed jobs. synchronous jobs are naturally only reported on completion and go directly to this collection.
* Updates to request models following changes in version-manager terminology
* Updated docs for docker compose for mongodb

### CODE REVIEW

File: job_service/api/request_models.py [hunk 1]
```diff
@@ -7,9 +7,9 @@
 
 
 class CommandEnum(str, Enum):
-    ADD_DATASET = 'ADD_DATA'
-    DELETE_DATASET = 'REMOVE'
-    CHANGE_DATASET = 'CHANGE_DATA'
+    ADD_OR_CHANGE_DATA = 'ADD_OR_CHANGE_DATA'
+    REMOVE_DATASET = 'REMOVE'
+    CHANGE_DATASET = 'IMPORT'
```
Comment (pawbu) [inline job_service/api/request_models.py:12]: Remove?

File: job_service/repository/job_db.py [hunk 2]
```diff
@@ -16,25 +17,39 @@
...
-            password=MONGODB_PASSWORD
+            password=MONGODB_PASSWORD,
+            authSource="admin"
+        )
+        db = client.jobdb
+        self.completed = db.completed
+        self.in_progress = db.inprogress
+        # Creates an unique index on datasetName for inprogress collection
+        # if index not already present.
+        self.in_progress.create_index(
```
Comment (pawbu) [inline job_service/repository/job_db.py:35]: 👍

**Comment by DanielElisenberg** [comment]
@pawbu @vaskalan I posted this a tad too early. I need to edit some of the unit tests.

File: job_service/repository/job_db.py [hunk 3]
```diff
@@ -65,65 +83,71 @@ def new_job(self, command: str, status: str,
...
+
+        if status in ["done", "failed"]:
+            self.in_progress.delete_one(find_query)
+            self.completed.insert_one(job)
+            self.completed.update_one(find_query, update_status_query)
+            if log is not None:
+                self.completed.update_one(find_query, add_log_query)
+            return
+
+        self.in_progress.update_one(find_query, update_status_query)
```
Comment (pawbu) [inline job_service/repository/job_db.py:147]: Updating `in_progress` list when the job might have been deleted with `delete_one`?

File: job_service/repository/job_db.py [hunk 4]
```diff
@@ -65,65 +83,71 @@ def new_job(self, command: str, status: str,
...
-                        "logs": {"at": now, "message": log}
-                    }
-                }
-            )
+            self.in_progress.update_one(find_query, add_log_query)
 
     def __stringify_datetime_logs(self, logs):
         for log in logs:
-            log["at"] = log["at"].strftime("%Y/%m/%d, %H:%M:%S")
+            log["at"] = log["at"].isoformat()
```
Comment (pawbu) [inline job_service/repository/job_db.py:153]: 💯

File: job_service/api/request_models.py (hunk 1 above)
Comment (DanielElisenberg) [reply job_service/api/request_models.py:12]: is REMOVE not the correct command terminology for setting a dataset to pending_delete?

File: job_service/repository/job_db.py (hunk 2 above)
Comment (DanielElisenberg) [reply job_service/repository/job_db.py]: This sadly caused issues with unit tests and the dependency tree :( I updated the database setup docs to include creating an index on the inprogress collection. The index info is stored in the persistent directory we will do backup on, so it should cause no issues to do it this way 👍🏻

File: job_service/repository/job_db.py (hunk 3 above)
Comment (DanielElisenberg) [reply job_service/repository/job_db.py:142]: if it enters the if statement that moves it from inprogress to completed the function terminates on the return on line 145. I can make this an if/else instead if that makes it more explicit?

Comment (pawbu) [reply job_service/repository/job_db.py]: It's OK, I didn't see the return statement :)

File: README.md [hunk 5]
```diff
@@ -2,8 +2,22 @@
 Service for managing jobs in the microdata platform.
 
 
-## Contribute
+## Database
+This service is bundled with a instance of [mongodb run through docker](https://hub.docker.com/_/mongo), and uses two collections under the jobDb database:
```
Comment (pawbu) [inline README.md:6]: Like the explanation 👍

File: job_service/repository/job_db.py [hunk 6]
```diff
@@ -65,65 +78,71 @@ def new_job(self, command: str, status: str,
...
+            raise NotFoundException(f"Could not find job with id {job_id}")
+
+        if status in ["done", "failed"]:
+            self.in_progress.delete_one(find_query)
+            self.completed.insert_one(job)
+            self.completed.update_one(find_query, update_status_query)
+            if log is not None:
+                self.completed.update_one(find_query, add_log_query)
+            return
+        else:
```
Comment (pawbu) [inline job_service/repository/job_db.py:141]: True, this is more explicit 👍

File: test/api/test_job_api.py [hunk 7]
```diff
@@ -11,7 +11,7 @@
     {'job_id': 'abc-abc-abc-abc'}
 ]
 NEW_JOB_REQUEST = {
-    "command": "ADD_DATA",
+    "command": "ADD_OR_CHANGE_DATA",
```
Comment (pawbu) [inline test/api/test_job_api.py:14]: 💯

File: test/resources/local/docker-compose.yml [hunk 8]
```diff
@@ -1,25 +1,26 @@
...
 #
 # poetry run gunicorn job_service.app:app
 
 version: '2'
 
 services:
-  mongodb:
-    image: docker.io/bitnami/mongodb:4.4
+  mongodbdb:
+    image: mongo:5
```
Comment (pawbu) [inline test/resources/local/docker-compose.yml:18]: 👍

**Review by pawbu — APPROVED** [review]
Looks good!

**Pull request merged**